build-logic/convention/src/main/kotlin/JavaConvention.kt

```diff
@@ -11,7 +11,7 @@ import java.net.URI
 class JavaConvention : Plugin<Project> {
   override fun apply(target: Project) {
     with(target) {
-      target.version = target.findProperty("VERSION_NAME")?.toString() ?: target.version
+      target.version = providers.gradleProperty("VERSION_NAME").orNull ?: target.version
```

Comment:
This isn't needed to fix the issues for consumers but I figured I'd fix the ewtf issues too while I'm here.